packages/rspack/src/rspack.ts

```diff
@@ -111,6 +112,32 @@ function revalidateWithStrategy(options: RspackOptions | MultiRspackOptions) {
 	}
 }
 
+function revalidateTsconfig(options: RspackOptions | MultiRspackOptions) {
```

Comment:
IMO, we should only detect `tsconfig.json` while using CLI. I'm not sure whether should have this logic in `@rspack/core`.

@hardfist cc

There is a similar feature at https://github.com/web-infra-dev/rspack/pull/2575

Reply:
So do I need to put my code in the cli?

Reply:
@hardfist cc

Reply:
we shouldn't put this logic into core, @rspack/core should align with webpack

Reply:
I refactored the code and put it in @rspack/cli cc @hardfist